config/v1alpha1/types_insights.go

```diff
@@ -0,0 +1,73 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// InsightsDataGather provides data gather configuration options for the the Insights Operator.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type InsightsDataGather struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	Spec InsightsDataGatherSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status InsightsDataGatherStatus `json:"status"`
+}
+
+type InsightsDataGatherSpec struct {
+	// gatherConfig spec attribute includes all the configuration options related to
+	// gathering of the Insights data and its uploading to the ingress.
+	// +optional
+	GatherConfig GatherConfig `json:"gatherConfig,omitempty"`
+}
+
+type InsightsDataGatherStatus struct {
+}
+
+// gatherConfig provides data gathering configuration options.
+type GatherConfig struct {
+	// dataPolicy allows user to enable additional global obfuscation of the IP addresses and base domain
+	// in the Insights archive data. Valid values are "None" and "IPsAndClusterDomain".
+	// When set to None the data is not obfuscated.
+	// When set to IPsAndClusterDomain the IP addresses and the cluster domain name are obfuscated.
+	// No value equals the "None" policy.
+	// +optional
+	DataPolicy DataPolicy `json:"dataPolicy,omitempty"`
+	// disabledGatherers is a list of gatherers to be excluded from the gathering. All the gatherers can be disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
```

Comment:
the API is ok, but where does a user locate the list of possible gatherers?

Reply:
This question has occured in https://github.com/openshift/enhancements/pull/1037#discussion_r847663887. I still think listing all the gatherers here is not a good approach (because it will likely change). Can we refer to `oc get insightsoperators.operator.openshift.io cluster -o json | jq .status.gatherStatus.gatherers` ?

Reply:
I'm just hoping to get this doc updated to tell me where to find the list.  I'd be ok with such a command.

Reply:
> // The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.

Line 46 has a link to a doc with the IDs, that should be roughly sufficient but a command is even better